cmake/developer_package/compile_flags/os_flags.cmake

```diff
@@ -208,6 +258,46 @@ macro(ov_arm_neon_fp16_optimization_flags flags)
     endif()
 endmacro()
 
+#
+# ov_arm_sve_optimization_flags(<output flags>)
+#
+macro(ov_arm_sve_optimization_flags flags)
+    if(OV_COMPILER_IS_INTEL_LLVM)
+        message(WARNING "Unsupported CXX compiler ${CMAKE_CXX_COMPILER_ID}")
+    elseif(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+        # nothing to define; works out of box
+    elseif(ANDROID)
+        if(ANDROID_ABI STREQUAL "arm64-v8a")
+            set(${flags} -march=armv8-a+sve -Wno-unused-command-line-argument)
+        else()
+            message(WARNING "SVE is not supported on this Android ABI: ${ANDROID_ABI}")
+        endif()
+    else()
+        if(AARCH64)
+            ov_check_compiler_supports_sve(CXX "-march=armv8-a+sve")
+            ov_check_compiler_supports_sve(C "-march=armv8-a+sve")
```

Comment:
should we perform this check for ANDROID as well? I'm sure Android toolchain supports SVE starting from specific compiler version, while currently you add the `-march=armv8-a+sve` flag unconditionally

Reply:
I've added the check as suggested. As per [this source](https://developer.android.com/ndk/guides/abis), I can only see SVE being supported on `arm64-v8a` as of now. Let me know if the checks are not correctly implemented.